examples/complete/variables.tf

```diff
@@ -103,6 +103,24 @@ variable "arn_format" {
   default     = "aws"
 }
 
+variable "create_dynamodb_table" {
+  type        = bool
+  description = "Boolean for if a DynamoDB table is to be created for batch locking."
+  default     = false
+}
+
+variable "user_managed_dynamodb_table_name" {
```

Comment:
Sure, that makes sense 👍

Reply:
But if this has a default value, we wouldn't be able to do the above condition right?

```
locals {
  dynamodb_table = (
    var.create_dynamodb_table            ? aws_dynamodb_table.geff_batch_locking_table[0]: 
    var.user_managed_dynamodb_table_name ? data.aws_dynamodb_table.user_managed_table[0]:
                                           null
  )
}
```

Reply:
yes, and that seems like the ideal default, since the perspective of most users is ignorance about the 30s limitation of API Gateway having an impact on potential workloads, and all the need to know is that this creates a dynamodb table needed for some kind of batch synchronization mechanisms and go on with their installation. the few users that want to give it a custom name or not have a backend db or make a custom should be allowed to customize it, but most users should get it without having to worry about what it is.

on that thought, we might want to call it something more general, to support other features (e.g. rate limiting) in the future, without having to rename the table, maybe `_batch_sync`?